src/dftbp/geoopt/rationalfunc.F90

```diff
@@ -130,6 +135,30 @@ subroutine step(this, val, grad, displ)
   end subroutine step
 
 
+  !> Reset optimizer
+  subroutine reset(this)
+
+    !> Instance of geometry optimization driver
+    class(TRationalFunc), intent(inout) :: this
+
+    integer :: ii, nvar1, npvar, npvar1
+
+    nvar1 = this%nvar + 1
```

Comment:
Could you add a comment as to why/what this is doing

Reply:
Better not, otherwise I would have to write something like "superfluous temporary integers". Jokes aside, I have removed the unnecessary lines, including the same copy&paste error in the step() function.